src/tools/clippy/clippy_lints/src/drop_forget_ref.rs

```diff
@@ -203,13 +95,8 @@ impl<'tcx> LateLintPass<'tcx> for DropForgetRef {
             && let Some(fn_name) = cx.tcx.get_diagnostic_name(def_id)
         {
             let arg_ty = cx.typeck_results().expr_ty(arg);
-            let is_copy = is_copy(cx, arg_ty);
             let drop_is_single_call_in_arm = is_single_call_in_arm(cx, arg, expr);
             let (lint, msg) = match fn_name {
-                sym::mem_drop if arg_ty.is_ref() && !drop_is_single_call_in_arm => (DROP_REF, DROP_REF_SUMMARY),
-                sym::mem_forget if arg_ty.is_ref() => (FORGET_REF, FORGET_REF_SUMMARY),
-                sym::mem_drop if is_copy && !drop_is_single_call_in_arm => (DROP_COPY, DROP_COPY_SUMMARY),
-                sym::mem_forget if is_copy => (FORGET_COPY, FORGET_COPY_SUMMARY),
```

Comment:
We'd want to keep these around but as an early return to avoid the extra cases appearing in e.g. `clippy::drop_non_drop`

https://github.com/rust-lang/rust/pull/109732/files#diff-ac6c6b2cb3727ed4c68d901f79187bfc83a4958814e261c425714e8319c29915R1-R13